### DISCUSSION TITLE
connection opened to posthog.com and app.posthog.com

### DISCUSSION THREAD

**Discussion by diolwe** [lukevella/rallly#584] [Q&A] [answered]
rallly opens a connection to posthog.com and app.hosthog.com ... to do what? 
How do I switch this off??

**Comment by lukevella** [comment]
Hi @diolwe, please use discussions instead of issues for questions. I've gone ahead and moved this one for you.

To answer your question, posthog is used to help analyze trends and usage on rallly.co. It's not intended for use by self-hosters but there doesn't seem to be a way to conditionally initialize it without it at least trying to open a connection. [See the posthog docs](https://posthog.com/tutorials/multiple-environments#conditional-initialization) where they recommend using a fake token to conditionally initialize. I'm not a fan of this either but to my understanding, it's harmless and shouldn't pose a risk. No data is being sent to posthog in your instance because it is not initialized.

There probably is a better way to handle this but it's not something I'm prioritizing right now since it makes no difference in reality for anyone. If a case was made that this posed some kind of security risk then it would be given a higher priority.

**Comment by lukevella** [comment, marked as answer]
Hi @diolwe, coincidentally I ended up revising this part of the code base and managed to find a solution to avoid this request from being made in environments were posthog is not intended to run. This patch was released in [v2.5.1](https://github.com/lukevella/rallly/releases/tag/v2.5.1).

**Comment by diolwe** [comment]
Jeeez! Where is your buy-me-a-coffee button?? I'll test this as soon as possible but thank you very much in advance!

**Comment by diolwe** [comment]
A dream came true! Thank you very much again. Coffee should be on its way.

**Reply by lukevella** [reply]
Thanks @diolwe, much appreciated! ☕️
